src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,155 @@
+            currentNode.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T value : list) {
+            add(value);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNodeByIndex(index);
+        return currentNode.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkIndex(index);
+        Node<T> currentNode = findNodeByIndex(index);
+        T oldValue = currentNode.item;
+        currentNode.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> nodeToRemove = findNodeByIndex(index);
+        unlink(nodeToRemove);
+        return nodeToRemove.item;
     }
 
     @Override
     public boolean remove(T object) {
+        Node<T> objectToRemove = first;
+        for (int i = 0; i < size; i++) {
+            if (object == objectToRemove.item
+                    || (object != null && object.equals(objectToRemove.item))) {
+                unlink(objectToRemove);
+                return true;
+            }
+            objectToRemove = objectToRemove.next;
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void unlink(Node<T> nodeToRemove) {
+        if (nodeToRemove == first && first.next == null) {
+            first = null;
+        } else if (nodeToRemove == first) {
+            first.next.prev = null;
+            first = first.next;
+        } else if (nodeToRemove == last) {
+            nodeToRemove.prev.next = null;
+            nodeToRemove.prev = last;
+        } else {
+            nodeToRemove.next.prev = nodeToRemove.prev;
+            nodeToRemove.prev.next = nodeToRemove.next;
+        }
+        size--;
+    }
+
+    private void checkIndexAdd(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayIndexOutOfBoundsException(
+                    String.format(INDEX_OUT_OF_BOUNDS_MESSAGE, index, size));
+        }
+    }
+
+    private void checkIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayIndexOutOfBoundsException(
+                    String.format(INDEX_OUT_OF_BOUNDS_MESSAGE, index, size));
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        Node<T> foundNode = first;
+        for (int i = 0; i < index; i++) {
+            foundNode = foundNode.next;
+        }
+        return foundNode;
+    }
+
+    private static class Node<T> {
+        private T item;
+        private Node<T> next;
+        private Node<T> prev;
+
+        Node(T item) {
+            this.item = item;
+        }
```

Comment:
![image](https://github.com/mate-academy/jv-linked-list/assets/156338484/60b853dd-95f2-4fbc-b5c0-8c9bef9d3e67)
Try to use this constructor